pill-press-app/Controllers/ApplicationController.cs

```diff
@@ -939,12 +939,109 @@ public async Task<IActionResult> DownloadCertificate(string id)
                 }
                 else
                 {
-                    _logger.LogError(LoggingEvents.HttpGet, "Unable to get account from application."); 
+                    _logger.LogError(LoggingEvents.HttpGet, "Unable to get account from application.");
                     return BadRequest();
                 }
-                
+            }
+            else
+            {
+                return BadRequest();
+            }
+        }
+        [HttpGet("{id}/certificate-exists")]
+        public async Task<IActionResult> CertificateExists(string id)
+        {
+            _logger.LogDebug(LoggingEvents.HttpPost, "Begin method " + this.GetType().Name + "." + MethodBase.GetCurrentMethod().ReflectedType.Name);
+            bool fileExists = false;
 
-                
+            if (!string.IsNullOrEmpty(id) && Guid.TryParse(id, out Guid applicationId))
+            {
+
+                // verify the currently logged in user has access to this account
+
+                MicrosoftDynamicsCRMincident application = _dynamicsClient.GetApplicationByIdWithChildren(applicationId);
+                if (application == null)
+                {
+                    _logger.LogWarning(LoggingEvents.NotFound, "Application NOT found.");
+                    return new NotFoundResult();
+                }
+
+                if (!UserDynamicsExtensions.CurrentUserHasAccessToApplication(applicationId, _httpContextAccessor, _dynamicsClient))
+                {
+                    _logger.LogWarning(LoggingEvents.NotFound, "Current user has NO access to the application.");
+                    return new NotFoundResult();
+                }
+
+                string applicationTypeName = application.BcgovApplicationTypeId.BcgovName;
+                string filePrefix = "";
+
+                /*             
+                 * File name format is
+                 * WA_CERTIFICATE_<CERTIFICATE_NUMBER> (for Waiver)
+                 * RS_CERTIFICATE_<CERTIFICATE_NUMBER> (for Registered Seller)
+                 * EC_CERTIFICATE_<CERTIFICATE_NUMBER> (For Equipment Notification)
+                 * 
+                 */
+
+                if (Guid.TryParse(application._customeridValue, out Guid accountId))
```

Comment:
I am concerned about this duplication of code - could you refactor this so that there is a function that generates the filename, and use it here and also the download action?